scalafix-tests/unit/src/test/scala/scalafix/tests/cli/CliGitDiffTests.scala

```diff
@@ -0,0 +1,242 @@
+package scalafix.tests.cli
+
+import java.lang.ProcessBuilder
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+import java.nio.charset.StandardCharsets
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Paths, Path, StandardOpenOption}
+
+import org.scalactic.source.Position
+import org.scalatest._
+import org.scalatest.FunSuite
+
+import scala.collection.immutable.Seq
+import scala.util._
+import scalafix.cli
+import scalafix.internal.cli.CommonOptions
+import scalafix.testkit.DiffAssertions
+
+class CliGitDiffTests() extends FunSuite with DiffAssertions {
+  gitTest("it should handle addition") { (fs, git, cli) =>
+    val oldCode = "old.scala"
+    val newCode = "new.scala"
+    val newCodeAbsPath = fs.absPath(newCode)
+
+    fs.add(
+      oldCode,
+      """|object OldCode {
+         |  // This is old code, where var's blossom
+         |  var oldVar = 1
+         |}""".stripMargin)
+    git.add(oldCode)
+    addConf(fs, git)
+    git.commit()
+
+    git.checkout("pr-1")
+
+    fs.add(
+      newCode,
+      """|object NewCode {
+         |  // New code, no vars
+         |  var newVar = 1
+         |}""".stripMargin)
+    git.add(newCode)
+    git.commit()
+
+    val obtained = runDiff(cli)
+
+    val expected =
+      s"""|Running DisableSyntax
+          |$newCodeAbsPath:3:3: error: [DisableSyntax.keywords.var] keywords.var is disabled
+          |  var newVar = 1
+          |  ^
+          |""".stripMargin
+
+    assertNoDiff(obtained, expected)
+  }
+
+  gitTest("it should handle modification") { (fs, git, cli) =>
+    val oldCode = "old.scala"
+    val oldCodeAbsPath = fs.absPath(oldCode)
+
+    fs.add(
+      oldCode,
+      """|object OldCode {
+         |  // This is old code, where var's blossom
+         |  var oldVar = 1
+         |}""".stripMargin)
+    git.add(oldCode)
+    addConf(fs, git)
+    git.commit()
+
+    git.checkout("pr-1")
+    fs.replace(
+      oldCode,
+      """|object OldCode {
+         |  // This is old code, where var's blossom
+         |  var oldVar = 1
+         |}
+         |object NewCode {
+         |  // It's not ok to add new vars
+         |  var newVar = 2
+         |}""".stripMargin
+    )
+    git.add(oldCode)
+    git.commit()
+
+    val obtained = runDiff(cli)
+
+    val expected =
+      s"""|Running DisableSyntax
+          |$oldCodeAbsPath:7:3: error: [DisableSyntax.keywords.var] keywords.var is disabled
+          |  var newVar = 2
+          |  ^
+          |""".stripMargin
+
+    assertNoDiff(obtained, expected)
+  }
+
+  gitTest("it should handle rename") { (fs, git, cli) =>
```

Comment:
Can we add a test for unstaged/staged files? From my experience in scalafmt, we should treat those as new files. It's convenient to be able to run scalafix on changes before commit

Reply:
git diff will only pick staged files with `git diff --cached` is this what you want?